File: kotlin/programmers/src/전화번호목록.java
import java.util.*;


class 전화번호목록 {
    public boolean solution(String[] phoneBook) {
        Map<String, Integer> map = new HashMap<>();

        HashMap<Integer , Integer> x = new HashMap<Integer , Integer>();
        x.put(3 , x.getOrDefault(2 , 0)+1);


        for (int i = 0; i < phoneBook.length; i++) map.put(phoneBook[i], i);
        for (int i = 0; i < phoneBook.length; i++)
            for (int j = 0; j < phoneBook[i].length(); j++)
                if (map.containsKey(phoneBook[i].substring(0, j))) return false;
        return true;
    }
}

File: kotlin/programmers/src/전화번호목록.kt
class Solution {
    val hashMap = hashMapOf<Char, MutableList<String>>()
    fun solution(phone_book: Array<String>): Boolean {
        val answer = true
        phone_book.forEach { phoneBookStr ->
            hashMap[phoneBookStr[0]]?.forEach { hashmapStr ->
                if (phoneBookStr < hashmapStr && phoneBookStr.startsWith(hashmapStr)) return false
                else if (phoneBookStr > hashmapStr && hashmapStr.startsWith(phoneBookStr)) return false
                else hashMap[phoneBookStr[0]]?.add(hashmapStr)
            } ?: run { hashMap[phoneBookStr[0]] = mutableListOf<String>(phoneBookStr) }
        }
        return answer
    }
}

